src/styles/variables.js

```diff
@@ -47,5 +69,14 @@ export default {
     tooltipzIndex: 10050,
     gutterWidth: 16,
     popoverMenuShadow: '0px 4px 12px 0px rgba(0, 0, 0, 0.06)',
+    lineHeightSmall: getValueUsingPixelRatio(14, 16),
+    lineHeightNormal: getValueUsingPixelRatio(16, 21),
+    lineHeightLarge: getValueUsingPixelRatio(18, 24),
+    lineHeightXLarge: getValueUsingPixelRatio(20, 22),
+    inputHeight: getValueUsingPixelRatio(52, 72),
+    pickerIconTop: getValueUsingPixelRatio(16, 26),
+    formErrorLineHeight: getValueUsingPixelRatio(18, 23),
+    communicationsLinkHeight: getValueUsingPixelRatio(20, 30),
+    alternateTextHeight: getValueUsingPixelRatio(20, 22),
```

Comment:
The code is improved a bit, but we are still exporting static values? Don't we want these to be dynamic?

Reply:
_Our App behavior should be:_

If the user tries to change the device font size while the App is opening (in the background), nothing will happen until the user completely close and reopen the App again, this is the same as what [YouTube does](https://github.com/Expensify/App/pull/8782#issuecomment-1134520093), and any App that doesn’t follow this approach will face the bug I mentioned [here](https://github.com/Expensify/App/pull/8782#discussion_r901977836), and [here](https://github.com/Expensify/App/pull/8782#issuecomment-1142399330).


we only need to get the current font scale when the App started, once it started we didn’t need to know more about the current font scale. The only time we need to get the current font scale while the App is opening is in `BaseHTMLEngineProvider.js` to stop any HTML inside `RenderHTMLConfigProvider` from increasing/decreasing as we explained [here](https://github.com/Expensify/App/pull/8782#discussion_r889129455), So we don’t want these to be dynamic.

Reply:
This is really the only blocker left from me. Looks good otherwise.

Reply:
>Ok so we only want it to be dynamic in the RenderHTMLConfigProvider... ?

Yes.

>And that is because the RenderHTMLConfigProvider is doing something to resize the text automatically?

Yes, as it explained in [this stackoverflow answer](https://stackoverflow.com/questions/41807843/how-to-disable-font-scaling-in-react-native-for-ios-app)  
`some components wont obey font scaling settings, for example: Alert, PickerIOS, DatePickerIOS, TabBarIOS, SegmentedControlIOS as these are all natively drawn and don't rely on the Text component.`

Although we are disallowing font scaling in our App, these components `<TRenderEngineProvider> <RenderHTMLConfigProvider>` inside `BaseHTMLEngineProvider.js` will not obey our font scaling settings and will keep increasing/decreasing if the user try to increase/decrease the device font size while the App is opening, So we need to stop these from increase/decrease manually, this is why we are function in this case.

Reply:
@marcaaron are we okay with this PR I still feel like do nothing for now and explore better options.

Reply:
@Santhosh-Sellavel I'm just trying to understand why we have it and don't get it. Can you explain it in your own words?

Reply:
My bad, I found that we can use `defaultTextProps` inside `RenderHTMLConfigProvider` component to disallow font scaling and it works fine, now we don't need `appStateListener` , `state` or functions anymore.

>how can the user increase/decrease the font size while the app is opening?

I mean sending the App to the background (not completely closing it), changing the device font size, and then resuming the App as in this video : 

https://user-images.githubusercontent.com/51829206/175270877-07538391-e726-4964-a1c2-1da62be3ce35.mov

Reply:
> @Santhosh-Sellavel I'm just trying to understand why we have it and don't get it. Can you explain it in your own words?

@marcaaron 
What are we trying to do here?
We try to restrict the font scale of the text throughout the app until the app is restarted(closing and reopening the app). But however, this `RenderHTMLConfigProvider` component doesn’t bound to the restriction made here.  So any text rendered using `RenderHTMLConfigProvider` component will be dynamically resized.

Why are we doing this change in `RenderHTMLConfigProvider`?
Just to stop this dynamic resizing for the text rendered by `RenderHTMLConfigProvider`, like the rest of the text in the app.